test_maxnrows is unstable

Occasionally, `test_maxnrows_on_large_dataset` fails for no good reason:
```
    def test_maxnrows_on_large_dataset():
        # Limit number of threads during reading, otherwise the test may fail on a
        # machine with many cores (all chunks read at once, in the same amount of
        # time as the single chunk containing the first 5 rows).
        src = b"A,B,C\n" + b"\n".join(b"%06d,x,1" % i for i in range(2000000))
        t0 = time.time()
        d0 = dt.fread(src, nthreads=4, max_nrows=5, verbose=True)
        t0 = time.time() - t0
        d0.internal.check()
        assert d0.shape == (5, 3)
        assert d0.topython() == [[0, 1, 2, 3, 4], ["x"] * 5, [True] * 5]
        t1 = time.time()
        d1 = dt.fread(src, nthreads=4, verbose=True)
        t1 = time.time() - t1
        assert d1.shape == (2000000, 3)
>       assert t0 < t1 / 2, ("Reading with max_nrows=5 should be faster than "
                             "reading the whole dataset")
E       AssertionError: Reading with max_nrows=5 should be faster than reading the whole dataset
E       assert 0.02736210823059082 < (0.05139899253845215 / 2)
```
The failure is probably caused by the test machine doing something else in the background during this test, which caused the read times to be longer than usual.

Proposed solution: if the condition `t0 < t1 /2` fails, try doing the test again several times.

--- a/tests/fread/test_fread_small.py
+++ b/tests/fread/test_fread_small.py
@@ -1109,6 +1109,13 @@ def test_maxnrows_on_large_dataset():
     d1 = dt.fread(src, nthreads=4, verbose=True)
     t1 = time.time() - t1
     assert d1.shape == (2000000, 3)
+    n_attempts = 3
+    while n_attempts > 0 and t0 >= t1 / 2:
+        time.sleep(0.5)
+        t0 = time.time()
+        d0 = dt.fread(src, nthreads=4, max_nrows=5)
+        t0 = time.time() - t0
+        n_attempts -= 1
     assert t0 < t1 / 2, ("Reading with max_nrows=5 should be faster than "
                          "reading the whole dataset")
 

